src/relay/op/tensor/transform.cc

```diff
@@ -0,0 +1,77 @@
+/*!
+ *  Copyright (c) 2018 by Contributors
+ * \file transform.cc
+ * \brief Transformation operators
+ */
+#include <tvm/relay/op.h>
+#include <tvm/relay/attrs/image.h>
+#include <vector>
+#include "../nn/layout.h"
+
+namespace tvm {
+namespace relay {
+
+// Flatten
+bool FlattenRel(const Array<Type>& types,
+               int num_inputs,
+               const Attrs& attrs,
+               const TypeReporter& reporter) {
+  CHECK_EQ(types.size(), 2);
+  const auto* data = types[0].as<TensorTypeNode>();
+  if (data == nullptr) return false;
+  if (data->shape.size() == 0) return false;
+
+  auto target_dim = make_const(Int(32), 1);
```

Comment:
Will it be better to use `Int(64)`?

Reply:
Done.

Reply:
Default type is Int(32) for all indexing till now.
Using Int(64) require explicit cast on the result shape (TensorType) comparison.

Reply:
@srkreddy1238 I see. Thanks!